src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,114 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    dst_f8->f1[0] = __logf(fabsf(src_f8->f1[0]));
+    dst_f8->f1[1] = __logf(fabsf(src_f8->f1[1]));
+    dst_f8->f1[2] = __logf(fabsf(src_f8->f1[2]));
+    dst_f8->f1[3] = __logf(fabsf(src_f8->f1[3]));
+    dst_f8->f1[4] = __logf(fabsf(src_f8->f1[4]));
+    dst_f8->f1[5] = __logf(fabsf(src_f8->f1[5]));
+    dst_f8->f1[6] = __logf(fabsf(src_f8->f1[6]));
+    dst_f8->f1[7] = __logf(fabsf(src_f8->f1[7]));
+
+    if constexpr (std::is_same<T, schar>::value)
+    {
+        dst_f8->f4[0] = rpp_hip_pixel_check_0to255(dst_f8->f4[0]) - (float4)128;
+        dst_f8->f4[1] = rpp_hip_pixel_check_0to255(dst_f8->f4[1]) - (float4)128;
```

Comment:
For I8 case, output is thresholded to range of I8 case?
For U8 case, is it f32 and retained as is?

Reply:
This was a mistake, removed this.